### ISSUE TITLE
The mAP results are quite different between model.predict and run_test=True

### ISSUE THREAD

**Issue by gzhex** [roboflow/rf-detr#496]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

I have set run_test=True when training, and it reported the mAP@50 of testset is 50. 
But when I used model.predict() to predict testset, and use supervision to calculate the mAP@50, it is only 18.
What's wrong with it?
Is there any difference between **model.predict()** and **redetr.engine.evaluate()** (run_test=True) ?

### Environment

RF-DETR: 1.3.0
OS: Ubuntu 20.04
Python: 3.10.0
PyTorch: 2.3.1
GPU: NVIDIA RTX 4090

### Minimal Reproducible Example

1. Just like readme, model.train(run_test=True). mAP=50
2. Just like readme, detections = model.predict(). And use supervision to calculate. mAP=18

### Additional

<img width="984" height="446" alt="Image" src="https://github.com/user-attachments/assets/f877d1f8-44cb-47a7-9d95-3611184fd2bf" />

<img width="941" height="176" alt="Image" src="https://github.com/user-attachments/assets/c1a7a425-4177-441d-af20-6376e79e8a0b" />

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by isaacrob-roboflow** [comment]
Are you using segmentation model?

**Comment by gzhex** [comment]
> I ask because supervision doesn't support masking mAP, which may be the source of the discrepancy. we use pycocotools to produce mAP and supervision is intended to produce the exact same result, so if there is a discrepancy there is likely an implementation issue with supervision. other than that it is hard to imagine why there would be a gap between predict and the run_test method 😅

I use the object detection model. I also think it is not support supervison, then I used pycocotools to calculate mAP. It is same as supervison. 
And I think it is caused by DDP. Then I used one GPU to do, still exists a gap.

Maybe the image preprocess is different? Between use predict() and evaluate()(run_test==True)

**Comment by isaacrob-roboflow** [comment]
Can you send a minimum reproducible example with a trained model showing this behavior?
